master/internal/trial.go

```diff
@@ -426,6 +426,16 @@ func (t *trial) maybeAllocateTask() error {
 		if err != nil {
 			return err
 		}
+
+		// persist the allocation workspace/experiment record, in the event of moves or deletions
+		err = task.InsertTrialAllocationWorkspaceRecord(
+			context.Background(), t.experimentID, ar.AllocationID)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to insert allocation workspace record when recovering allocation: %w",
```

Comment:
same as above comment, user gets a failed msg, but the allocation is running?

Reply:
Fair point, it's worth shifting this up so that we can persist this information even if the allocation fails while also failing the trial/task if it's unable to store it's workspace info